File: fix_buttons.dart
import 'dart:io';

void main() {
  final dir = Directory('lib/screens');
  final files = dir.listSync(recursive: true).whereType<File>().where((f) => f.path.endsWith('.dart'));
  
  for (final file in files) {
    String content = file.readAsStringSync();
    String original = content;

    // 1. Remove explicit shapes from ElevatedButton, FilledButton, OutlinedButton, TextButton
    // We match shape: RoundedRectangleBorder(...)
    final shapeRegex1 = RegExp(r'shape:\s*RoundedRectangleBorder\([^)]*\),?');
    final shapeRegex2 = RegExp(r'shape:\s*const\s*RoundedRectangleBorder\([^)]*\),?');
    
    // For specific radius variables like AppRadius.xs
    final shapeRegex3 = RegExp(r'shape:\s*[^,]*AppRadius\.(xs|sm|md|lg|button)[^,]*,?');

    content = content.replaceAll(shapeRegex1, '');
    content = content.replaceAll(shapeRegex2, '');
    content = content.replaceAll(shapeRegex3, '');

    // 2. Inject AppRadius.button into InkWell
    // Replace InkWell( with InkWell(borderRadius: AppRadius.button, 
    content = content.replaceAll('InkWell(', 'InkWell(borderRadius: AppRadius.button, ');

    // Now we might have duplicate borderRadius in InkWell, e.g. 
    // InkWell(borderRadius: AppRadius.button, borderRadius: BorderRadius.circular(10))
    // Or InkWell(borderRadius: AppRadius.button, ... borderRadius: AppRadius.xs)
    // We will clean up the old borderRadius: ... by removing it.
    // We want to remove the SECOND borderRadius if it exists.
    // Since regex across multiple lines is tricky, we'll do this:
    // First, temporarily change our injected one to a unique string:
    content = content.replaceAll('InkWell(borderRadius: AppRadius.button, ', '@@@INKWELL@@@');
    
    // Now any remaining borderRadius inside InkWell needs to be removed.
    // Actually, this is getting complicated because borderRadius could be anywhere.
    // Let's do something simpler: replace known borderRadius patterns in the file
    content = content.replaceAll(RegExp(r'borderRadius:\s*BorderRadius\.circular\([^)]*\),?'), '');
    content = content.replaceAll(RegExp(r'borderRadius:\s*AppRadius\.(xs|sm|md|lg|button|pill),?'), '');

    // Restore the injected InkWell
    content = content.replaceAll('@@@INKWELL@@@', 'InkWell(borderRadius: AppRadius.button, ');

    // 3. Make sure AppRadius is imported if we injected AppRadius.button
    if (content.contains('AppRadius.button') && !content.contains('app_radius.dart')) {
       // Find the last import and insert after it
       final importIndex = content.lastIndexOf(RegExp(r'import\s+[^;]+;'));
       if (importIndex != -1) {
          final insertIndex = content.indexOf('\n', importIndex) + 1;
          content = content.substring(0, insertIndex) + "import 'package:stayhub_mobile/theme/app_radius.dart';\n" + content.substring(insertIndex);
       }
    }

    if (content != original) {
      file.writeAsStringSync(content);
      print('Updated ${file.path}');
    }
  }
}

File: fix_buttons.js
const fs = require('fs');
const path = require('path');

function walkDir(dir, callback) {
    fs.readdirSync(dir).forEach(f => {
        const dirPath = path.join(dir, f);
        const isDirectory = fs.statSync(dirPath).isDirectory();
        isDirectory ? walkDir(dirPath, callback) : callback(path.join(dir, f));
    });
}

function processFile(filePath) {
    if (!filePath.endsWith('.dart')) return;
    let content = fs.readFileSync(filePath, 'utf8');
    let original = content;

    // 1. Remove shape: RoundedRectangleBorder(...) inside styleFrom()
    // By using [\s\S]*? we match multiple lines carefully
    const buttonRegex = /(ElevatedButton|FilledButton|OutlinedButton|TextButton)\.styleFrom\s*\(([\s\S]*?)\)/g;
    
    content = content.replace(buttonRegex, (match, btnType, styleArgs) => {
        // We only want to remove `shape: ...` if it's there
        let newArgs = styleArgs.replace(/shape:\s*(const\s+)?RoundedRectangleBorder\([^)]*\),?/g, '');
        // Also remove if it's using AppRadius directly (e.g. AppRadius.xs)
        newArgs = newArgs.replace(/shape:\s*(const\s+)?[^,]+AppRadius\.[a-z]+[^,]*,?/g, '');
        
        // Return the reconstructed styleFrom
        return `${btnType}.styleFrom(${newArgs})`;
    });

    // 2. InkWell without borderRadius
    // Let's find InkWell( and inject borderRadius: AppRadius.button,
    // BUT only if it doesn't already have one inside its own parameter list
    let index = 0;
    while ((index = content.indexOf('InkWell(', index)) !== -1) {
        let openParenCount = 1;
        let i = index + 'InkWell('.length;
        let start = i;
        
        while (i < content.length && openParenCount > 0) {
            if (content[i] === '(') openParenCount++;
            if (content[i] === ')') openParenCount--;
            i++;
        }
        
        let inkWellBody = content.substring(start, i - 1);
        
        if (!inkWellBody.includes('borderRadius:')) {
            content = content.substring(0, start) + 'borderRadius: AppRadius.button, ' + content.substring(start);
            index = start + 'borderRadius: AppRadius.button, '.length;
        } else {
            // It has borderRadius. Let's fix it if it's not AppRadius.button.
            // Replace `borderRadius: BorderRadius.circular(...)` with `borderRadius: AppRadius.button`
            // Replace `borderRadius: AppRadius.xs` with `borderRadius: AppRadius.button`
            let updatedBody = inkWellBody.replace(/borderRadius:\s*BorderRadius\.circular\([^)]*\),?/g, 'borderRadius: AppRadius.button,');
            updatedBody = updatedBody.replace(/borderRadius:\s*AppRadius\.[a-z]+,?/g, 'borderRadius: AppRadius.button,');
            content = content.substring(0, start) + updatedBody + content.substring(i - 1);
            index = start + updatedBody.length + 1;
        }
    }

    if (content !== original) {
        // Check if we need to import app_radius.dart
        if (content.includes('AppRadius.button') && !content.includes('app_radius.dart')) {
            let imports = content.match(/import\s+[^;]+;/g);
            if (imports && imports.length > 0) {
                let lastImport = imports[imports.length - 1];
                let lastImportIndex = content.lastIndexOf(lastImport) + lastImport.length;
                content = content.substring(0, lastImportIndex) + "\nimport 'package:stayhub_mobile/theme/app_radius.dart';" + content.substring(lastImportIndex);
            } else {
                content = "import 'package:stayhub_mobile/theme/app_radius.dart';\n" + content;
            }
        }
        fs.writeFileSync(filePath, content, 'utf8');
        console.log(`Updated ${filePath}`);
    }
}

walkDir('lib/screens', processFile);
